### PR TITLE
[Feature-16269][dinky-task]Supports dinky tasks to accept Prepare Params passed within a task or from a previous task

### LINKED ISSUE #16269: [Feature][dinky task] Supports dinky tasks to accept Prepare Params passed within a task or from a previous task
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/dolphinscheduler/issues?q=is%3Aissue) and found no similar feature requirement.

### Description

Supports dinky tasks to accept variables passed within a task or from a previous task

Just like this
image

### Use case

_No response_

### Related issues

_No response_

### Are you willing to submit a PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by gaoyan1998**
## Purpose of the pull request

Supports dinky tasks to accept variables passed within a task or from a previous task

close #16269

### CODE REVIEW

**Review by SbloodyS — COMMENTED** [review]
Please supplement the document. @gaoyan1998

**Comment by gaoyan1998** [comment]
> Please supplement the document. @gaoyan1998

@SbloodyS PTAL, Supplement the documentation done.

**Comment by SbloodyS** [comment]
Please run `mvn spotless:apply` to format the code. @gaoyan1998

**Comment by gaoyan1998** [comment]
> Please run `mvn spotless:apply` to format the code. @gaoyan1998

I have resolved it，please review it，thanks. @SbloodyS

File: dolphinscheduler-task-plugin/dolphinscheduler-task-dinky/src/main/java/org/apache/dolphinscheduler/plugin/task/dinky/DinkyTask.java [hunk 1]
```diff
@@ -343,15 +344,30 @@ private Map<String, String> generateVariables() {
             }
         }
         List<Property> localParams = this.dinkyParameters.getLocalParams();
+        Map<String, Property> prepareParamsMap = taskExecutionContext.getPrepareParamsMap();
         if (localParams == null || localParams.isEmpty()) {
             return variables;
         }
+        Map<String, String> convertMap = convert(prepareParamsMap);
```
Comment (ruanwenjun) [inline dolphinscheduler-task-plugin/dolphinscheduler-task-dinky/src/main/java/org/apache/dolphinscheduler/plugin/task/dinky/DinkyTask.java]:
```suggestion
        Map<String, String> convertMap = ParameterUtils.convert(prepareParamsMap);
```

File: dolphinscheduler-task-plugin/dolphinscheduler-task-dinky/src/main/java/org/apache/dolphinscheduler/plugin/task/dinky/DinkyTask.java [hunk 2]
```diff
@@ -343,15 +344,30 @@ private Map<String, String> generateVariables() {
...
+        if (paramsMap == null) {
+            return null;
+        }
+        Map<String, String> map = new HashMap<>();
+        for (Map.Entry<String, Property> en : paramsMap.entrySet()) {
+            map.put(en.getKey(), en.getValue().getValue());
+        }
+        return map;
+    }
+
```
Comment (ruanwenjun) [inline dolphinscheduler-task-plugin/dolphinscheduler-task-dinky/src/main/java/org/apache/dolphinscheduler/plugin/task/dinky/DinkyTask.java]:
```suggestion

```

**Review by ruanwenjun — CHANGES_REQUESTED** [review]

**Comment by gaoyan1998** [comment]
@ruanwenjun Thanks for your suggestion, I have finished the revision

**Review by ruanwenjun — DISMISSED** [review]
LGTM

**Comment by gaoyan1998** [comment]
hi @ruanwenjun PTAL ,please approval workflows

**Review by ruanwenjun — APPROVED** [review]
LGTM

**Review by SbloodyS — COMMENTED** [review]
Please create an issue and link to it. @gaoyan1998

**Comment by gaoyan1998** [comment]
> Please create an issue and link to it. @gaoyan1998

@SbloodyS Sorry for the missing issue, it has now been created and the title changed

**Review by SbloodyS — APPROVED** [review]
+1

**Pull request merged**